src/lib/offline-api/index.ts

```diff
@@ -485,6 +500,68 @@ class OfflineAPI {
     return this.requestBatches
   }
 
+  async createTag (id: string): Promise<Tag> {
+    this.assertRecording()
+
+    const tagData: APITag = {
+      sys: {
+        id,
+        version: 0
+      },
+      name: undefined
+    }
+
+    const tag = new Tag(tagData)
+
+    this.modifiedTags.set(id, tag)
+
+    return tag
+  }
+
+  async saveTag (id: string) {
+    this.assertRecording()
+
+    const hasTag = await this.hasTag(id)
+
+    if (!hasTag) {
+      throw new Error(`Cannot save the tag (id: ${id}) because it does not exist`)
+    }
+
+    const tag = await this.getTag(id)
+    // Store clone as a request
+    this.currentRequestsRecorded.push(saveTagRequest(tag.clone()))
+
+    // Mutate version bump
+    tag.version = tag.version + 1
+
+    this.modifiedTags.set(id, tag)
+
+    // TODO Do we need the savedTags?
+    // this.savedTags.set(id, tag.clone())
+
+    // TODO What is validated here?
```

Comment:
I think here we'd validate the final shape of the payload against the schema.

So you'd define a schema for what the tags payload looks like (I guess right now just `name` being a string between X and Y characters long)  and validate those here if I recall correctly.

Reply:
I implemented this second round of payload validations for tags. I am not sure if this makes too much sense for now however. As far as I understand and suspect, this second round of payload validations was implemented to check for cases such as content types having too many fields at the end of a migration, something which you would not be able to check during the original create content type intent. In the case of tags however, the updates are far less complex, the only thing that needs to be checked is if there is a correct `name` property. But maybe it still makes sense to have this check here for the future, when editing tags becomes less trivial? Or am I missing something?